### PR TITLE
Forgot Password

### PR DESCRIPTION
**Pull request by yosefanajjar**
issue: #14 **Forgot Password**
## What did I do?
- I made the forgot password page 
- I made it responsive and it looks good in most  screen sizes

## How to test?
Just go to the `/forgot-password` route and you shall see the page 

**Note**: there are things which need to be done in the future like making the firebase function

### CODE REVIEW

File: src/containers/ForgotPassword/index.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
-import UnderConstruction from '../UnderConstruction';
+import ForgotPassword from './forgotPass';
 
-const ForgotPassword = () => {
-  return <UnderConstruction />;
-};
+function index() {
+  return <ForgotPassword />;
+}
```
Comment (fares98) [inline src/containers/ForgotPassword/index.js]: use ES6

**Review by fares98 — CHANGES_REQUESTED** [review]

**Review by fares98 — DISMISSED** [review]

File: src/containers/ForgotPassword/index.js [hunk 2]
```diff
@@ -1,6 +1,41 @@
-import React from 'react';
+import React, { Component } from 'react';
 import ForgotPassword from './forgotPass';
+import schema from './utils/schema';
 
-const index = props => <ForgotPassword {...props} />;
+export default class index extends Component {
+  // eslint-disable-next-line react/state-in-constructor
```
Comment (Fatmasiam) [inline src/containers/ForgotPassword/index.js]: eslint-disable-next-line !!!! 
**WHY**

File: src/routes/App/style.css [hunk 3]
```diff
@@ -1,6 +1,14 @@
-@font-face {
-  font-family: 'VAGRounded';
-  src: url('../../components/assets/fonts/vagrounded-thin.ttf') format('truetype');
+@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,700&display=swap');
+@import url('https://fonts.googleapis.com/css?family=Poppins:400,500,700&display=swap');
```
Comment (Fatmasiam) [inline src/routes/App/style.css]: you can import both fonts in the same URL 
---
```suggestion
@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700|Poppins:300,400,500,700&display=swap');
```

File: src/containers/ForgotPassword/index.js [hunk 4]
```diff
@@ -1,8 +1,41 @@
-import React from 'react';
-import UnderConstruction from '../UnderConstruction';
+import React, { Component } from 'react';
+import ForgotPassword from './forgotPass';
```
Comment (Fatmasiam) [inline src/containers/ForgotPassword/index.js:1]: add a broken line between the module import and local module.
---
The best ordering for the import is :
1. standard library modules — e.g.`react`
    * `add a broken line then point 2--> `
2. third-party library modules (anything installed) — e.g.`yup`, `react-router-dom`, etc.
    * `add a broken line then point 3-->` 
3. locally-developed modules( our file that we working on it )

File: src/containers/ForgotPassword/forgotPass.css [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  background-color: #e9663aec;
+  border-color: #e9663aec;
+}
+
+.forgot-password .ant-btn-primary:hover,
+.forgot-password .ant-btn-primary:focus {
+  color: #fff;
+  background-color: #e9663aec;
+  border-color: #e9663aec;
+}
```
Comment (Fatmasiam) [inline src/containers/ForgotPassword/forgotPass.css:77]: repeated style !!! 
from line 74 to 76, and 81 to 83

File: src/containers/ForgotPassword/forgotPass.css [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+.forgot-password__button {
+  width: 80%;
```
Comment (Fatmasiam) [inline src/containers/ForgotPassword/forgotPass.css]: repeated ` width:80%` ?!  and line 50

**Comment by Fatmasiam** [comment]
## Please write a description for **WHAT** the _page do_ ,**Not** what you are done

**Comment by Fatmasiam** [comment]
Check the accessibility
---- 
Screenshot from 2019-10-31 21-08-48

**Comment by Fatmasiam** [comment]
**WHY** the page scrolling up and down ?!

File: src/containers/ForgotPassword/utils/schema.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import * as yup from 'yup';
```
Comment (Fatmasiam) [inline src/containers/ForgotPassword/utils/schema.js]: move the utils folder to the `src/` folder  root

File: src/containers/ForgotPassword/vector.svg [hunk 8]
```diff
@@ -0,0 +1,54 @@
+<svg width="212" height="157" viewBox="0 0 212 157" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (Fatmasiam) [inline src/containers/ForgotPassword/vector.svg]: move the `svg` image to the `assets/images/ ` folder

File: src/containers/ForgotPassword/index.js [hunk 9]
```diff
@@ -1,8 +1,41 @@
...
+      if (!valid) {
+        this.setState({ error: 'the input is not a valid email' });
+      } else {
+        this.setState({ error: '' });
+      }
+    });
+  };
+
+  handleClick = () => {
+    // Logic will be added hereW
```
Comment (Fatmasiam) [inline src/containers/ForgotPassword/index.js]:
```suggestion
    // Logic will be added here
```
it's will be good if you write a cleaner comment like: make a  fetch to firebase server :smiley:

File: src/containers/ForgotPassword/index.js [hunk 10]
```diff
@@ -1,8 +1,41 @@
...
+  };
+
+  render() {
+    const { email, error } = this.state;
+    const { ...props } = this.props;
```
Comment (Fatmasiam) [inline src/containers/ForgotPassword/index.js]: unnecessary line 
```suggestion
```

File: src/containers/ForgotPassword/index.js [hunk 11]
```diff
@@ -1,8 +1,41 @@
...
+    return (
+      <ForgotPassword
+        email={email}
+        error={error}
+        handleChange={this.handleChange}
+        handleClick={this.handleClick}
+        {...props}
```
Comment (Fatmasiam) [inline src/containers/ForgotPassword/index.js]:
```suggestion
        {...this.props}
```

File: src/containers/ForgotPassword/index.js [hunk 12]
```diff
@@ -1,8 +1,41 @@
...
+    error: '',
+  };
 
-export default ForgotPassword;
+  handleChange = ({ target: { value } }) => {
+    const { email } = this.state;
+    this.setState({ email: value }, async () => {
+      const valid = await schema.isValid({ email: email.trim() });
+      if (!valid) {
+        this.setState({ error: 'the input is not a valid email' });
```
Comment (Fatmasiam) [inline src/containers/ForgotPassword/index.js]:
```suggestion
        this.setState({ error: 'Please, enter valid email ' });
```

**Review by AlaaTaima — DISMISSED** [review]

**Review by fares98 — DISMISSED** [review]

File: src/containers/ForgotPassword/index.js [hunk 13]
```diff
@@ -1,8 +1,5 @@
 import React from 'react';
```
Comment (fares98) [inline src/containers/ForgotPassword/index.js]: remove this line

**Review by fares98 — CHANGES_REQUESTED** [review]

File: src/containers/ForgotPassword/forgotPass.jsx [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+  };
+
+  return (
+    <div className="forgot-password">
+      <header className="forgot-password__heading">
+        <div className="forgot-password__back-btn">
+          <BackButton handleBack={goBack} />
+        </div>
+        <h2 className="forgot-password__title">forgot password</h2>
+      </header>
```
Comment (fares98) [inline src/containers/ForgotPassword/forgotPass.jsx]: Use the Header component

**Review by fares98 — CHANGES_REQUESTED** [review]

File: src/containers/ForgotPassword/forgotPass.jsx (hunk 14 above)
Comment (fares98) [inline src/containers/ForgotPassword/forgotPass.jsx]: Use the header component

**Review by fares98 — CHANGES_REQUESTED** [review]

**Review by Fatmasiam — DISMISSED** [review]

File: src/components/Header/header.css [hunk 15]
```diff
@@ -2,52 +2,31 @@
...
 .signup__heading {
   font-weight: bold;
-  font-family: "VAGRounded", sans-serif !important;
-  width: 64px;
+  font-family: sans-serif !important;
+  width: max-content;
   margin: 0;
-  margin-right: calc(50vw - 32px);
+  text-transform: capitalize;
+  transform: translateX(-40%);
```
Comment (Fatmasiam) [inline src/components/Header/header.css]:
```suggestion
  transform: translateX(-58%);
```

**Review by Fatmasiam — APPROVED** [review]

**Review by AlaaTaima — APPROVED** [review]

**Review by fares98 — APPROVED** [review]

**Review by alaa-yasin — APPROVED** [review]

**Pull request merged**